src/daemons/StorageDaemon.cpp

```diff
@@ -72,27 +75,47 @@ int main(int argc, char *argv[]) {
     }
 
     if (FLAGS_data_path.empty()) {
-        LOG(FATAL) << "Storage Data Path should not empty";
-        return -1;
+        LOG(ERROR) << "Storage Data Path should not empty";
+        return EXIT_FAILURE;
+    }
+
+    auto result = nebula::network::NetworkUtils::getLocalIP(FLAGS_local_ip);
+    if (!result.ok()) {
+        LOG(ERROR) << "Get localIp failed, ip " << FLAGS_local_ip
+                   << ", status:" << result.status();
+        return EXIT_FAILURE;
+    }
+    auto hostRet = nebula::network::NetworkUtils::toHostAddr(result.value(), FLAGS_port);
+    if (!hostRet.ok()) {
+        LOG(ERROR) << "Bad local host addr, status:" << hostRet.status();
+        return EXIT_FAILURE;
+    }
+    auto& localHost = hostRet.value();
+    auto metaAddrsRet = nebula::network::NetworkUtils::toHosts(FLAGS_meta_server_addrs);
+    if (!metaAddrsRet.ok() || metaAddrsRet.value().empty()) {
+        LOG(ERROR) << "Can't get metaServer address, status:" << metaAddrsRet.status()
+                   << ", FLAGS_meta_server_addrs:" << FLAGS_meta_server_addrs;
+        return EXIT_FAILURE;
     }
-    LOG(INFO) << "Starting the storage Daemon on port " << FLAGS_port
-              << ", dataPath " << FLAGS_data_path;
 
     std::vector<std::string> paths;
     folly::split(",", FLAGS_data_path, paths, true);
     std::transform(paths.begin(), paths.end(), paths.begin(), [](auto& p) {
         return folly::trimWhitespace(p).str();
     });
-    auto result = nebula::network::NetworkUtils::getLocalIP(FLAGS_local_ip);
-    CHECK(result.ok()) << result.status();
-    uint32_t localIP;
-    CHECK(NetworkUtils::ipv4ToInt(result.value(), localIP));
+    if (paths.empty()) {
+         LOG(ERROR) << "Bad data_path format:" << FLAGS_data_path;
+         return EXIT_FAILURE;
+    }
 
-    auto metaClient = std::make_unique<nebula::meta::MetaClient>();
+    auto ioThreadPool = std::make_shared<folly::IOThreadPoolExecutor>(FLAGS_io_handlers);
```

Comment:
We could share the same thread pool with the one in ThriftServer

Reply:
I have done it.  Please note line 151

Reply:
Excellent.